contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinLido.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/LibERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+
+
+/// @dev Minimal interface for minting StETH
+interface ILido {
+    /// @dev Adds eth to the pool
+    /// @param _referral optional address for referrals
+    /// @return StETH Amount of shares generated
+    function submit(address _referral) external payable returns (uint256 StETH);
+    /// @dev Retrieve the current pooled ETH representation of the shares amount
+    /// @param _sharesAmount amount of shares
+    /// @return amount of pooled ETH represented by the shares amount
+    function getPooledEthByShares(uint256 _sharesAmount) external view returns (uint256);
+}
+
+
+contract MixinLido {
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibERC20TokenV06 for IEtherTokenV06;
+
+    IEtherTokenV06 private immutable WETH;
+
+    constructor(IEtherTokenV06 weth)
+        public
+    {
+        WETH = weth;
+    }
+
+    function _tradeLido(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256 boughtAmount)
+    {
+        (ILido lido) = abi.decode(bridgeData, (ILido));
+        if (address(sellToken) == address(WETH) && address(buyToken) == address(lido)) {
+            WETH.withdraw(sellAmount);
+            // TODO(kimpers): do we want to use a referral address here
+            boughtAmount = lido.getPooledEthByShares(lido.submit{ value: sellAmount}(address(0)));
```

Comment:
Do we want to set an affiliate address here for tracking purposes? 🤔

Reply:
I vote no. We don't do this anywhere else. It might potentially add gas as well?